script.module.clouddrive.common/clouddrive/common/ui/addon.py

```diff
@@ -300,11 +306,144 @@ def _list_drive(self, driveid):
                 if 'context_options' in folder:
                     list_item.addContextMenuItems(folder['context_options'])
                 listing.append((url, list_item, True))
+            if self._content_type == 'video' or self._content_type == 'audio':
+                url = self._addon_url + '?' + urllib.urlencode({'action':'_list_exports', 'content_type': self._content_type, 'driveid': driveid})
+                listing.append((url, xbmcgui.ListItem(self._common_addon.getLocalizedString(32000)), True))
+            
             xbmcplugin.addDirectoryItems(self._addon_handle, listing, len(listing))
             xbmcplugin.endOfDirectory(self._addon_handle, True)
         else:
             self._list_folder(driveid, path='/')
 
+    def _list_exports(self, driveid):
+        export_manager = ExportManager(self._account_manager._addon_data_path)
+        exports = export_manager.load()
+        listing = []
+        for exportid in exports:
+            export = exports[exportid]
+            if export['driveid'] == driveid and export['content_type'] == self._content_type:
+                item_name = Utils.unicode(export['name'])
+                params = {'action':'_open_export', 'content_type': self._content_type, 'driveid': driveid, 'item_driveid': export['item_driveid'], 'item_id': export['id'], 'name': urllib.quote(Utils.str(item_name))}
+                url = self._addon_url + '?' + urllib.urlencode(params)
+                list_item = xbmcgui.ListItem(item_name)
+                context_options = []
+                params['action'] = '_run_export'
+                context_options.append((KodiUtils.localize(21479), 'RunPlugin('+self._addon_url + '?' + urllib.urlencode(params)+')'))
+                params['action'] = '_remove_export'
+                context_options.append((KodiUtils.localize(1210), 'RunPlugin('+self._addon_url + '?' + urllib.urlencode(params)+')'))
+                list_item.addContextMenuItems(context_options)
+                listing.append((url, list_item, True))
+        xbmcplugin.addDirectoryItems(self._addon_handle, listing, len(listing))
```

Comment:
this stuff theoretically should be handled but the plugins that consume the module, not by the module itself

Reply:
This module is used as common library for cloud drive plugins. 
OneDrive and Google Drive addons (and others in the future) do the same logic in almost all the things. Only the access to its respective API changes. So each addon 'inherit' all the common logic from here and then do its respective stuff.
So for me it doesn't make sense to repeat the same code over and over when a concept of 'module' already exist within the kodi ecosystem.